web-common/src/features/dashboards/time-dimension-details/charts/tdd-tooltip-formatter.ts

```diff
@@ -41,21 +50,36 @@ export function tddTooltipFormatter(
       content += "<table>";
 
       const selectedValuesLength = selectedDimensionValues.length;
-      if (chartType === TDDChart.STACKED_AREA && selectedValuesLength) {
+      if (
+        chartType === TDDChart.STACKED_AREA &&
+        (selectedValuesLength || isTimeComparison)
+      ) {
         for (const key of keys) {
           const val = rest[key];
-
           if (val === null || val === "NaN") continue;
 
+          let label = key;
+          let keyColor = colorMap[String(key)] || "#000";
+
+          if (isTimeComparison) {
+            if (key.startsWith("comparison.")) {
+              keyColor = colorMap["comparison_period"];
+              label = "Previous Period";
+            } else {
+              keyColor = colorMap["current_period"];
+              label = "Current Period";
+            }
+          }
+
           content += `
             <tr>
               <td class="color">
                 <svg width="16" height="16">
-                  <circle cx="8" cy="8" r="6" style="fill:${colorMap[String(key)] || "#000"};">
+                  <circle cx="8" cy="8" r="6" style="fill:${keyColor};">
                   </reccit>
                 </svg>
               </td>
-              <td class="key">${escapeHTML(key)}</td>
+              <td class="key">${escapeHTML(label)}</td>
               <td class="value">${escapeHTML(String(val))}</td>
             </tr>`;
         }
```

Comment:
**Scalability Issue**: The dynamic generation of HTML content within the tooltip formatter function can lead to significant performance issues when rendering tooltips for large datasets, as each tooltip generation involves complex DOM manipulations. <br> **Fix**: Optimize the tooltip generation by pre-compiling templates or using lighter-weight DOM manipulation methods, especially for large datasets. <br> **Code Suggestion**: 
 ```
 +  const tooltipTemplate = '<tr><td class="color"><svg width="16" height="16"><circle cx="8" cy="8" r="6" style="fill:${keyColor};"></circle></svg></td><td class="key">${label}</td><td class="value">${value}</td></tr>';
 +  content += tooltipTemplate;
 ```